File: LeetCode_List/203Remove_Linked_List_Elements(final).py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

# 可以编译通过 但是runtime error


class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        if head==None:
        	return head
        head.next=self.removeElements(head.next,val):
        if head.val==val:
        	return head.next
        else:
        	return head

File: LeetCode_List/203Remove_Linked_List_Elements(final).cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

// 不要小瞧一个问题 当这个问题简单的时候 就是看如何用最精炼的代码 写出一个最为高效的代码

class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        if(head==NULL) { return head; }
        head->next=removeElements(head->next,val);
        return head.val==val ? head->next : head;
    }
};